### PR TITLE
fix(only-letters-textfield): prevent numbers in name fields

### PR DESCRIPTION
**Pull request by eladyashar**
create generic typePreventiveComponent
change all first name and last name to alphabetTextField that prevent typing of numbers

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { FormControl, Grid, InputLabel, MenuItem, Select, TextField } from '@mat
 
 import ContactType from 'models/ContactType';
 import FormInput from 'commons/FormInput/FormInput';
+import AlphabetTextField from 'commons/AlphabetTextField/AlphabetTextField';
 import AlphanumericTextField from 'commons/AlphanumericTextField/AlphanumericTextField';
```
Comment (EmilyLip) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx:9]: delete import

File: client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx [hunk 2]
```diff
@@ -14,52 +12,12 @@ const stringAlphanum = yup
...
-        value={value}
-        onChange={conditionalyTriggerOnChange}
-        onBlur={onBlur}
-        error={errorObject ? true : false}
-        label={errorObject ? errorObject.message : label}
-        placeholder={placeholder}
-        className={className}
-      />
-    </Tooltip>
+    <TypePreventiveTextField
```
Comment (EmilyLip) [inline client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx:16]: can we delete AlphanumericTextField file? (changed to AlphabetTextFieldType)

**Review by NaamanTsur — DISMISSED** [review]
Noice

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import PlacesTypesAndSubTypes from 'commons/Forms/PlacesTypesAndSubTypes/PlacesT
 
 import useFormStyles from 'styles/formStyles';
 import { useForm } from 'react-hook-form';
+import AlphabetTextField from 'commons/AlphabetTextField/AlphabetTextField';
 import AlphanumericTextField from 'commons/AlphanumericTextField/AlphanumericTextField';
```
Comment (EmilyLip) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx:11]: delete unused import

File: client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx (hunk 2 above)
Comment (NaamanTsur) [reply client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx:16]: There are two types: text field which only allows letters and a text field that allows letters and numbers.

Comment (EmilyLip) [reply client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx:16]: V

Comment (EmilyLip) [reply client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx:16]: so can we use TypePreventiveTextField ?

File: client/src/commons/TypePreventiveTextField/TypePreventiveTextField.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { TextField, Tooltip } from '@material-ui/core';
+
+import get from 'Utils/auxiliaryFunctions/auxiliaryFunctions'
+
+import TypePreventiveTextFieldType from './TypePreventiveTextFieldTypes';
+
+const TypePreventiveTextField: TypePreventiveTextFieldType = (props) => {
+  const value = (props.value == null || props.value === undefined) ? "" : props.value;
```
Comment (nevobonneh) [inline client/src/commons/TypePreventiveTextField/TypePreventiveTextField.tsx:9]: why not ===

File: client/src/commons/TypePreventiveTextField/TypePreventiveTextFieldTypes.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+export interface TypePreventiveTextFieldProps<T> {
+    name: string,
+    value: T | null,
+    onChange: (value: string) => void,
+    onBlur?: (event: React.ChangeEvent<{}>) => void,
+    required?: boolean,
+    label?: string,
+    setError: (name: string, error: { type?: string, types?: object, message?: string, shouldFocus?: boolean }) => void,
+    clearErrors: (name?: string | string[]) => void,
+    errors: any,
```
Comment (nevobonneh) [inline client/src/commons/TypePreventiveTextField/TypePreventiveTextFieldTypes.ts:13]: errors : Record<string, Object>

File: client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx (hunk 2 above)
Comment (NaamanTsur) [reply client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx:16]: TypePreventiveTextField is a generic component which accepts a `yup` schema and prevents typing according to it.
It was created so AlphanumericTextField and AlphabetTextField can be built from it. If there is a future need for more types of typing prevention we can now create components from TypePreventiveTextField to achieve that.

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx (hunk 1 above)
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/ContactForm/ContactForm.tsx:9]: i will

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx (hunk 3 above)
Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx:11]: i will

File: client/src/commons/TypePreventiveTextField/TypePreventiveTextFieldTypes.ts (hunk 5 above)
Comment (eladyashar) [reply client/src/commons/TypePreventiveTextField/TypePreventiveTextFieldTypes.ts:13]: done

File: client/src/commons/TypePreventiveTextField/TypePreventiveTextField.tsx (hunk 4 above)
Comment (eladyashar) [reply client/src/commons/TypePreventiveTextField/TypePreventiveTextField.tsx:9]: changed to !props.value ? "" : props.value;

File: client/src/commons/AlphabetTextField/AlphabetTextFieldTypes.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+export interface AlphabetTextFieldProps<T> {
+    name: string,
+    value: T | null,
+    onChange: (value: string) => void,
+    onBlur?: (event: React.ChangeEvent<{}>) => void,
+    required?: boolean,
+    label?: string,
+    setError: (name: string, error: { type?: string, types?: object, message?: string, shouldFocus?: boolean }) => void,
+    clearErrors: (name?: string | string[]) => void,
+    errors: any,
```
Comment (nevobonneh) [inline client/src/commons/AlphabetTextField/AlphabetTextFieldTypes.ts:12]: errors : Record<string, Object>

Comment (eladyashar) [reply client/src/commons/AlphabetTextField/AlphabetTextFieldTypes.ts]: done

**Review by nevobonneh — DISMISSED** [review]

**Review by nevobonneh — APPROVED** [review]

**Review by EmilyLip — DISMISSED** [review]

**Review by nevobonneh — APPROVED** [review]

**Review by EmilyLip — APPROVED** [review]

**Pull request merged**